addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/command/LegacyLiveDataUpdate.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.internal.command;
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.math.RoundingMode;
+import java.nio.charset.StandardCharsets;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.solaredge.handler.SolarEdgeHandler;
+import org.openhab.binding.solaredge.internal.callback.AbstractCommandCallback;
+import org.openhab.binding.solaredge.internal.model.LiveDataChannels;
+
+/**
+ * command that retrieves values for live data channels - legacy version for old setups
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public class LegacyLiveDataUpdate extends AbstractCommandCallback implements SolarEdgeCommand {
+    private static final String NCG_NON_GREEDY_TEXT_OR_NUMBER = "(?:.+?)";
+    private static final String CG_CURRENT_POWER_AND_UNIT = "(?:\\{currentPower:\"([0-9\\.,]+)(.+)\"\\})";
+    private static final String UNIT_W = "W";
+
+    private final SolarEdgeHandler handler;
+    private int retries = 0;
+
+    public LegacyLiveDataUpdate(SolarEdgeHandler handler) {
+        super(handler.getConfiguration());
+        this.handler = handler;
+    }
+
+    @Override
+    protected Request prepareRequest(Request requestToPrepare) {
+        requestToPrepare.followRedirects(false);
+        requestToPrepare.param(DATA_API_LEGACY_LIVE_DATA_FIELDID_FIELD, config.getSolarId());
+        requestToPrepare.method(HttpMethod.GET);
+
+        return requestToPrepare;
+    }
+
+    @Override
+    protected String getURL() {
+        return DATA_API_LEGACY_LIVE_DATA_URL;
+    }
+
+    @Override
+    public void onComplete(Result result) {
```

Comment:
This looks very similar to the other class, is more reuse possible?

Reply:
It is similar but it is not the same. It is only a few lines so should we really introduce more abstraction?

Reply:
No that is okay then.